src/coreclr/nativeaot/Runtime/CMakeLists.txt

```diff
@@ -24,6 +23,7 @@ set(COMMON_RUNTIME_SOURCES
     ObjectLayout.cpp
     portable.cpp
     RestrictedCallouts.cpp
+    RedhawkGCInterface.cpp
```

Comment:
Should we start moving away from the Redhawk name? One needs decoder ring to understand what it means.

What would this file be called if we have done further refactoring to share it with regular CoreCLR?

Reply:
> What would this file be called if we have done further refactoring to share it with regular CoreCLR?

I think EEToGC interface is a specifically a NativeAOT thing. It looks like there was a reason/effort in the past to not spread knowledge about GC all over the runtime. 

In CoreCLR it is more adhoc - we just do `GCHeapUtilities::GetGCHeap()` and call IGCHeap stuff. I do not think there is an interface. 
That is also what we do in many cases in NativeAOT now. `RedhawkGCInterface` is still convenient to park complicated helpers, but some `RedhawkGCInterface` methods just delegate to `GCHeapUtilities::GetGCHeap()`, so it could be more historical that we have this.

Maybe we do not need this interface. We could just call `GCHeapUtilities::GetGCHeap()` in simple cases and see what remains here. 
Possibly it will be just initialization and ref scanning, which may also go to places that use them - startup.cpp, thread.cpp and the like.

Reply:
> I think EEToGC interface is a specifically a NativeAOT thing. It looks like there was a reason/effort in the past to not spread knowledge about GC all over the runtime.

Yes, build of the native redhawk runtime was setup in very complicated way with a lot of artificial layers. I do not think we need to maintain that.

Reply:
Re: GCToEEInterface.cpp

I am not very attached to the actual name of this file. I had to chose some kind of name, preferrably self-documenting the purpose of the file, so I just matched the name of the class that is implemented there.

The coreclr does have an analog for this file - `gcenv.ee.cpp`. That is not very self-documenting name, but we could still use it here for consistency.

Reply:
I would use it here for consistency.